csharp/private/wrappers/dotnet.cc

```diff
@@ -50,6 +123,29 @@ int main(int argc, char** argv) {
       evprintf("DOTNET_CLI_TELEMETRY_OPTOUT", "1"),  // disable telemetry
   };
 
+  // variables available for substitution in arguments
+  auto workspaceDir = PWD();
+  std::map<std::string, std::string> substitutions = {
+#ifdef _WIN32
+      {"__BAZEL_WORKSPACE__", "\\"},
+#else   // not _WIN32
+      {"__BAZEL_WORKSPACE__", "/"},
+#endif  // _WIN32
+      {"__BAZEL_SANDBOX__", workspaceDir},
+  };
+
+  // variable substitution of inputs + params file
+  for (int i = 1; i < argc; i++) {
```

Comment:
We'll lose a lot of loops by just appending `/pathmap` rather than searching for it. It looks like we'd also support multiple of these kind of arguments but the motivation for that isn't clear to me.

Reply:
I don't expect to support more variables, just those that help scrub away baked in context like paths.

Reply:
So all I'm suggesting is to replace this with `"/pathmap:" + workspaceDir + "=" + dirSeperator`. The only extra thing the exe knows about is how `/pathmap` is formatted which seems perfectly fine: the job of the wrapper is to execute `dotnet.exe` inside Bazel and that's just part of it.